### ISSUE TITLE
0 percent branch coverage for classes

### ISSUE THREAD

**Issue by CodingBlob** [OpenCover/opencover#933]
### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET 4.7
- [ ] .NET 4.7.1
- [ ] .NET 4.7.2
- [ ] .NET Core 1.0.0
- [ ] .NET Core 2.0.0
- [x] Something else (.NET Core 3.1.0)

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [x] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

### I have already...

- [x] repeated the problem using the latest stable release of OpenCover.
- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [x] have looked at the opencover output xml file in an attempt to resolve the issue.
- [x] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [x] no coverage being recorded
- [ ] 32 or 64 bit support
- [ ] feature request

### Expected Behavior

The class should have 100% Branch coverage

### Actual Behavior

The class has 0% Branch coverage

### Steps to reproduce the problem:

I have a project targeting netstandard2.0 and net46 as target frameworks. The tests are run with dotnet test.
The test project has netcoreapp 3.1 as the target framework.

![Cover](https://user-images.githubusercontent.com/44905762/74732242-1f099280-524a-11ea-88c9-63ef82660119.PNG)

![Cover_Test](https://user-images.githubusercontent.com/44905762/74732079-b7ebde00-5249-11ea-9987-9b7e495b7561.PNG)

- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf) again because we know you probably didn't do it the first time.
- [x] sample code or link to repository/gist is available (support is much faster this way)

**Comment by lothrop** [comment]
Might be related to #931.

**Comment by PTwr** [comment]
I encountered something similar in .NET 4.6.1.

Branches in full if-else statement are detected correctly, but not in early-exit scenario although lines of code inside of conditional block are marked as visited.

![image](https://user-images.githubusercontent.com/20748035/75019050-7d27b700-5490-11ea-8308-27833102931c.png)

**Comment by shunter** [comment]
We saw a similar problem after updating NUnit.ConsoleRunner from version 3.10.0 to 3.11.1, after which many classes started reporting much lower or zero branch coverage (affecting a different subset of classes each time).  

We were able to work around this by running NUnit with --inprocess to disable multi-process test execution.  This doubles the runtime but gets our results working again.  

Given that it seems to affects a random subset of classes each time, and that disabling multi-process fixes it, this makes me suspect some sort of race condition in how OpenCover is reporting branch coverage information.  I looked at the release notes for NUnit.ConsoleRunner 3.11 and couldn't see any obvious changes that would be relevant to triggering this where the old version would not.
